### PR TITLE
feat(DocumentDB): New DocumentDB checks

### PR DESCRIPTION
**Pull request by sansns**
### Context
New Checks
1. Backup enabled
2. Cloudwatch Log Export for Profiler and Audit checks
3. Deletion Protection enabled

Modified Checks
1. Changed DocumentDB Instance storage encrypted check to check the Clusters storage encryption. 

### Description

Trend Conformity: https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/aws/DocumentDB/
AWS Security Hub: https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/documentdb/documentdb_cluster_storage_encrypted/documentdb_cluster_storage_encrypted.metadata.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "ResourceType": "AwsRDSDBCluster",
+  "Description": "Check if DocumentDB cluster storage is encrypted.",
+  "Risk": "Ensure that encryption of data at rest is enabled for your Amazon DocumentDB (with MongoDB compatibility) database clusters for additional data security and regulatory compliance.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-1",
+  "Remediation": {
+    "Code": {
+      "CLI": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/aws/DocumentDB/encryption-enabled.html#",
+      "NativeIaC": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/aws/DocumentDB/encryption-enabled.html#",
+      "Other": "",
+      "Terraform": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/aws/DocumentDB/encryption-enabled.html#"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_storage_encrypted/documentdb_cluster_storage_encrypted.metadata.json]:
```suggestion
      "Terraform": "https://docs.prowler.com/checks/aws/general-policies/bc_aws_general_28/"
```

File: prowler/providers/aws/services/documentdb/documentdb_cluster_storage_encrypted/documentdb_cluster_storage_encrypted.metadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
+  "Severity": "medium",
+  "ResourceType": "AwsRDSDBCluster",
+  "Description": "Check if DocumentDB cluster storage is encrypted.",
+  "Risk": "Ensure that encryption of data at rest is enabled for your Amazon DocumentDB (with MongoDB compatibility) database clusters for additional data security and regulatory compliance.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-1",
+  "Remediation": {
+    "Code": {
+      "CLI": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/aws/DocumentDB/encryption-enabled.html#",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_storage_encrypted/documentdb_cluster_storage_encrypted.metadata.json]:
```suggestion
      "CLI": "https://docs.prowler.com/checks/aws/general-policies/bc_aws_general_28/",
```

File: prowler/providers/aws/services/documentdb/documentdb_cluster_cloudwatch_log_export/documentdb_cluster_cloudwatch_log_export.metadata.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "documentdb_cluster_c_log_export",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_cloudwatch_log_export/documentdb_cluster_cloudwatch_log_export.metadata.json]:
```suggestion
  "CheckID": "documentdb_cluster_cloudwatch_log_export",
```

File: prowler/providers/aws/services/documentdb/documentdb_cluster_backup_enabled/documentdb_cluster_backup_enabled.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            report.resource_id = cluster.id
+            report.resource_arn = cluster.arn
+            report.resource_tags = cluster.tags
+            report.status = "FAIL"
+            report.status_extended = (
+                f"DocumentDB Cluster {cluster.id} does not have backup enabled."
+            )
+            if (
+                cluster.backup_retention_period > 0
+                and cluster.backup_retention_period < 7
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_backup_enabled/documentdb_cluster_backup_enabled.py]: Could you put this as a configurable variable, please?

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Thank you @sansns ! Please, check my comments.

**Comment by sansns** [comment]
Addressed all the comments.

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**